lib/CL/devices/almaif/AlmaifDB/DBDevice.hh

```diff
@@ -0,0 +1,69 @@
+/* DBDevice.hh - Device based on parsing Almaif database and instantiating
+ *               other device types based on what it finds from there
+
+   Copyright (c) 2023 Topi Leppänen / Tampere University
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to
+   deal in the Software without restriction, including without limitation the
+   rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+   sell copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+   IN THE SOFTWARE.
+*/
+
+#ifndef DBDEVICE_H
+#define DBDEVICE_H
+
+#include "../AlmaIFDevice.hh"
+#include "AlmaifDBManager.hh"
+
+class DBDevice : public AlmaIFDevice {
+
+public:
+  DBDevice(const std::string &db_path);
+  ~DBDevice();
+
+  virtual void loadProgramToDevice(almaif_kernel_data_s *kd, cl_kernel kernel,
+                                   _cl_command_node *cmd);
+  void printMemoryDump();
+  void writeDataToDevice(pocl_mem_identifier *dst_mem_id,
+                         const char *__restrict__ const src, size_t size,
+                         size_t offset) override;
+  void readDataFromDevice(char *__restrict__ const dst,
+                          pocl_mem_identifier *src_mem_id, size_t size,
+                          size_t offset) override;
+  cl_int allocate_buffer(pocl_mem_identifier *p, size_t size) override;
```

Comment:
This mixes C and C++ naming styles. We should use C++ naming style here, right?

Reply:
done